From e5126a59d822c00ad291c65e1a82681adee65758 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:24:36 -0400
Subject: [cpufreq] don't restore policy->cpus on failure to move kobj

Message-id: <1409235879-1283-183-git-send-email-prarit@redhat.com>
Patchwork-id: 89439
O-Subject: [RHEL7.1 PATCH BZ 1134639 182/185] cpufreq: don't restore policy->cpus on failure to move kobj
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit 1461dc7d1c252fbeb5baee0c94b5b28b7966e125
Author: Viresh Kumar <viresh.kumar@linaro.org>
Date:   Thu Jul 17 10:48:26 2014 +0530

    cpufreq: don't restore policy->cpus on failure to move kobj

    While hot-unplugging policy->cpu, we call cpufreq_nominate_new_policy_cpu() to
    nominate next owner of policy, i.e. policy->cpu. If we fail to move policy
    kobject under the new policy->cpu, we try to update policy->cpus with the old
    policy->cpu.

    This would have been required in case old-CPU is removed from policy->cpus in
    the first place. But its not done before calling
    cpufreq_nominate_new_policy_cpu(), but during the POST_DEAD notification which
    happens quite late in the hot-unplugging path.

    So, this is just some useless code hanging around, get rid of it.

    Signed-off-by: Viresh Kumar <viresh.kumar@linaro.org>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/cpufreq.c b/drivers/cpufreq/cpufreq.c
index 9d844bb..d02e730 100644
--- a/drivers/cpufreq/cpufreq.c
+++ b/drivers/cpufreq/cpufreq.c
@@ -1320,11 +1320,6 @@ static int cpufreq_nominate_new_policy_cpu(struct cpufreq_policy *policy,
  ret = kobject_move(&policy->kobj, &cpu_dev->kobj);
  if (ret) {
   pr_err("%s: Failed to move kobj: %d\n", __func__, ret);
-
-  down_write(&policy->rwsem);
-  cpumask_set_cpu(old_cpu, policy->cpus);
-  up_write(&policy->rwsem);
-
   ret = sysfs_create_link(&cpu_dev->kobj, &policy->kobj,
      "cpufreq");
 
-- 
1.7.1